Source/extensions/processcontainers/ContainerAdministrator.cpp

```diff
@@ -52,17 +52,50 @@ namespace ProcessContainers {
             container.Add(value.Data(), &specific);
             container.FromString(configuration);
 
+            if ((specific.IsSet() == false) && (_producers.size() == 1)) {
+                // Looks like the configuration for this provider is not specified, however there is only one container
+                // system available, so let's forward the entire config (perhaps it's a legacy config file).
+                specific = configuration;
+            }
+
             TRACE(Trace::Information, (_T("Initializing container runtime system '%s'..."), value.Data()));
 
             // Pass the configuration to the runtime
-            const uint32_t initResult = runtime.second->Initialize(specific);
+            const uint32_t initResult = runtime.second->Initialize(specific.Value());
 
             if (initResult != Core::ERROR_NONE) {
                 TRACE(Trace::Error, (_T("Initialization failure")));
                 result = Core::ERROR_GENERAL;
             }
         }
 
+        if (_producers.empty() == true) {
+            TRACE(Trace::Error, (_T("No container runtime systems are available!")));
```

Comment:
Shouldn't we make this a syslog? This is a message for the operator it configured the system incorrectly right?